redis/src/cluster_async/mod.rs

```diff
@@ -1145,22 +1217,92 @@ where
         }
     }
 
+    // Validate all existing user connections and try to reconnect if nessesary.
+    // In addition, as a safety measure, drop nodes that do not have any assigned slots.
+    // This function serves as a cheap alternative to slot_refresh() and thus can be used much more frequently.
+    // The function does not discover the topology from the cluster and assumes the cached topology is valid.
+    // In addition, the validation is done by peeking at the state of the underlying transport w/o overhead of additional commands to server.
+    async fn validate_all_user_connections(inner: Arc<InnerCore<C>>) {
+        let mut all_valid_conns = HashMap::new();
+        let mut all_nodes_with_slots = HashSet::new();
+        // prep connections and clean out these w/o assigned slots, as we might have established connections to unwanted hosts
+        {
+            let mut nodes_to_delete = Vec::new();
```

Comment:
better use `with_capacity`

Reply:
unfortunately the capacity is not known until the later stage and allocating with the maximum is "unlikely" required. We expect here only rare nodes

Reply:
Yea, I figured that the actual size will probably less than the max, but unless the number is differs by order of magnitude, it is still better allocating the space needed once rather having it increased for every `push` call

Reply:
we do not expect here more than a few so the default should be definitely enough